background/services/provider-bridge/db.ts

```diff
@@ -1,25 +1,25 @@
+import { PermissionRequest } from "@tallyho/provider-bridge-shared"
 import Dexie from "dexie"
 
-/**
- * An object describing a set of permissions granted to a given provider.
- * Generally tracked by dAppID as part of DAppPermissions.
- */
-export type ProviderPermissions = {
-  allowedAccounts: string[]
-}
-
-type DAppPermissions = {
-  dAppID: string
-  permissions: ProviderPermissions
-}
-
 type Migration = {
   id: number
   appliedAt: number
 }
 
+function keyBy(
+  permissionsArray: Array<PermissionRequest>,
+  key: keyof PermissionRequest
+): Record<string, PermissionRequest> {
+  return permissionsArray.reduce((previous, current) => {
+    // Reduce is meant to be used like this.
+    /* eslint-disable no-param-reassign */
+    previous[current[key]] = current
+    return previous
```

Comment:
Reduce can be used like this, but is not meant to :) The airbnb style guide explicitly requires either a special name (e.g. `acc`) or non-mutating updates, e.g.:

```suggestion
    // Reduce is meant to be used like this.
    /* eslint-disable no-param-reassign */
    previous[current[key]] = current
    return {
      ...previous,
      [current[key]]: current
    }
```

Reply:
Can you please ellaborate on the reasons behind the non-mutating updates _here_? 

As per my current understanding it's meant to protect the initial object from being modified or if the internal steps shared their object instance.

I changed the variable name to `acc` so our linter is happy now.
I chose previous as I thought it to be more descriptive/easier to read, but `acc` is what I usually use.

Just for reference: my initial implementation :) 
`then((p) => p.reduce((a, c) => ((a[c.origin] = c), a), {}))`